crates/db_schema/src/impls/password_reset_request.rs

```diff
@@ -52,23 +32,27 @@ impl PasswordResetRequest {
       local_user_id: from_local_user_id,
       token: token_,
     };
-
-    Self::create(pool, &form).await
+    let conn = &mut get_conn(pool).await?;
+    insert_into(password_reset_request)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
   }
-  pub async fn read_from_token(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
+
+  /// Reads reset token and invalidates it
+  pub async fn read(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    password_reset_request
+    update(password_reset_request)
+      .filter(valid.eq(true))
       .filter(token.eq(token_))
       .filter(published.gt(now.into_sql::<Timestamptz>() - 1.days()))
```

Comment:
This is the only line you need to change. Change this to a shorter time span, and none of this is necessary.

Reply:
Thats not it, having the reset token valid for a day is fine in case you dont receive it immediately. But it should be possible to reuse the same one multiple times.

Reply:
Alright, seems a bit overkill to me still, but its fine.